docs/internals/optimizer.rst

```diff
@@ -458,6 +472,12 @@ is transformed to
 This transformation can also be useful when paired with LoopInvariantCodeMotion, since
 invariants in the loop-invariant conditions can then be taken outside the loop.
 
+Loops that already have a literal constant as iteration condition are not transformed.
+
+To avoid unnecessary rewriting, it is recommended to run this step after StructuralSimplifier.
```

Comment:
How so, wouldn't the structural simplifier just remove for loops that have a `false` condition with its initialization? These cases would be ignored by the `ForLoopConditionIntoBody` step anyways